symphony-bdk-core/src/main/java/com/symphony/bdk/core/service/MessageService.java

```diff
@@ -1,64 +1,339 @@
+        new TemplateResolver());
+  }
 
-  public MessageService(MessagesApi messagesApi, AuthSession authSession) {
+  public MessageService(MessagesApi messagesApi, MessageApi messageApi, MessageSuppressionApi messageSuppressionApi,
+      StreamsApi streamsApi, PodApi podApi, AttachmentsApi attachmentsApi, DefaultApi defaultApi, AuthSession authSession,
+      TemplateEngine templateEngine) {
+    this(messagesApi, messageApi, messageSuppressionApi, streamsApi, podApi, attachmentsApi, defaultApi, authSession,
+        new TemplateResolver(templateEngine));
+  }
+
+  private MessageService(MessagesApi messagesApi, MessageApi messageApi, MessageSuppressionApi messageSuppressionApi,
+      StreamsApi streamsApi, PodApi podApi, AttachmentsApi attachmentsApi, DefaultApi defaultApi, AuthSession authSession,
+      TemplateResolver templateResolver) {
     this.messagesApi = messagesApi;
+    this.messageApi = messageApi;
+    this.messageSuppressionApi = messageSuppressionApi;
+    this.streamsApi = streamsApi;
+    this.podApi = podApi;
+    this.attachmentsApi = attachmentsApi;
     this.authSession = authSession;
+    this.templateResolver = templateResolver;
+    this.defaultApi = defaultApi;
+  }
+
+  /**
+   * Get messages from an existing stream. Additionally returns any attachments associated with the message.
+   *
+   * @param streamId the streamID where to look for messages
+   * @param since instant of the earliest possible date of the first message returned.
+   * @param skip number of messages to skip. Optional and defaults to 0.
+   * @param limit maximum number of messages to return. Optional and defaults to 50.
+   * @return the list of matching messages in the stream.
+   * @see <a href="https://developers.symphony.com/restapi/reference#messages-v4">Messages</a>
+   */
+  public List<V4Message> getMessages(@Nonnull String streamId, @Nonnull Instant since, Integer skip, Integer limit) {
+    return callAndCatchApiException(() -> messagesApi.v4StreamSidMessageGet(streamId, getEpochMillis(since),
+        authSession.getSessionToken(), authSession.getKeyManagerToken(), skip, limit));
+  }
+
+  /**
+   * Get messages from an existing stream. Additionally returns any attachments associated with the message.
+   *
+   * @param streamId the streamID where to look for messages
+   * @param since instant of the earliest possible date of the first message returned.
+   * @param chunkSize size of elements to retrieve in one call. Optional and defaults to 50.
+   * @param totalSize total maximum number of messages to return. Optional and defaults to 50.
+   * @return a {@link Stream} of matching messages in the stream.
+   * @see <a href="https://developers.symphony.com/restapi/reference#messages-v4">Messages</a>
+   */
+  public Stream<V4Message> getMessagesStream(@Nonnull String streamId, @Nonnull Instant since, Integer chunkSize, Integer totalSize) {
```

Comment:
does it make sense to still have the parameters chunkSize and totalSize if we return a stream that can return all the data?